### ISSUE TITLE
3.9.2 not working with code-server running in Kubernetes via Helm

### ISSUE THREAD

**Issue by jsjoeio** [cdr/code-server#3004]
_Reporting on behalf of @nodesocket_
## OS/Web Information

- Web Browser: Chrome 89.0.4389.90/Firefox 87.0
- Local OS: MacOS Big Sur
- Remote OS: Docker / Kubernetes using official container 3.9.2.
- Remote Architecture:
- `code-server --version`: 3.9.2

## Steps to Reproduce

1. Upgrade Kubernetes images from 3.9.1 to 3.9.2.
2. Then logged into my code-server and seeing these `WebSocket close with status code 1006`

## Expected

code-server `3.9.2` should work

## Actual

![image](https://user-images.githubusercontent.com/3806031/113028538-3cdd4280-9140-11eb-866c-9f33a930a68d.png)

```js
ERR [remote-connection][ExtensionHost][47c0b…][initial][<hidden>:80] socketFactory.connect() failed or timed out. Error:
log.ts:296   ERR Error: WebSocket close with status code 1006
    at WebSocket.<anonymous> (browserSocketFactory.ts:129)
```

```js
browserSocketFactory.ts:162 WebSocket connection to 'wss://<hidden>/?reconnectionToken=2cb654bd-29f1-4132-b584-f156eb9a4df6&reconnection=false&skipWebSocketFrames=false' failed: 
create @ browserSocketFactory.ts:162
connect @ browserSocketFactory.ts:212
s @ remoteAgentConnection.ts:192
r @ remoteAgentConnection.ts:233
i @ remoteAgentConnection.ts:310
o @ remoteAgentConnection.ts:334
w @ remoteAgentConnection.ts:407
log.ts:296   ERR [remote-connection][Management   ][2cb65…][initial][<hidden>:80] socketFactory.connect() failed or timed out. Error:
log.ts:296   ERR Error: WebSocket close with status code 1006
    at WebSocket.<anonymous> (browserSocketFactory.ts:129)
log.ts:296   ERR [remote-connection] An error occurred in the very first connect attempt, it will be treated as a permanent error! Error:
log.ts:296   ERR Error: WebSocket close with status code 1006
    at WebSocket.<anonymous> (browserSocketFactory.ts:129)
log.ts:296   ERR Error: WebSocket close with status code 1006
    at WebSocket.<anonymous> (browserSocketFactory.ts:129)
log.ts:290  WARN Ignoring the error while validating workspace folder vscode-remote:/<hidden>/home/coder - WebSocket close with status code 1006
webWorkerExtensionHost.ts:125 The web worker extension host is started without an iframe sandbox!
start @ webWorkerExtensionHost.ts:125
Kn @ extensionHostManager.ts:67
_createInstance @ instantiationService.ts:110
createInstance @ instantiationService.ts:76
(anonymous) @ abstractExtensionService.ts:447
_startExtensionHosts @ abstractExtensionService.ts:446
_initialize @ abstractExtensionService.ts:411
(anonymous) @ extensionService.ts:74
Promise.then (async)
Xs @ extensionService.ts:74
_createInstance @ instantiationService.ts:110
_createServiceInstance @ instantiationService.ts:230
_createServiceInstanceWithOwner @ instantiationService.ts:219
_createAndCacheServiceInstance @ instantiationService.ts:208
_safeCreateAndCacheServiceInstance @ instantiationService.ts:151
_getOrCreateServiceInstance @ instantiationService.ts:135
get @ instantiationService.ts:54
initLayout @ layout.ts:255
(anonymous) @ workbench.ts:137
invokeFunction @ instantiationService.ts:61
startup @ workbench.ts:130
open @ web.main.ts:95
async function (async)
open @ web.main.ts:83
Se @ web.main.ts:359
g @ workbench.web.api.ts:498
(anonymous) @ workbench.ts:498
(anonymous) @ workbench.ts:505
r._invokeFactory @ loader.js:1114
r.complete @ loader.js:1124
r._onModuleComplete @ loader.js:1750
r._onModuleComplete @ loader.js:1762
r._resolve @ loader.js:1710
r.defineModule @ loader.js:1353
o @ loader.js:1638
v @ loader.js:2145
r._invokeFactory @ loader.js:1114
r.complete @ loader.js:1124
r._onModuleComplete @ loader.js:1750
r._onModuleComplete @ loader.js:1762
r._resolve @ loader.js:1710
r.defineModule @ loader.js:1353
p @ loader.js:1800
(anonymous) @ workbench.nls.js:7
```

## Related

- https://github.com/cdr/code-server/issues/2975#issuecomment-808595111
- https://github.com/cdr/code-server/issues/2975#issuecomment-808611903
- https://github.com/cdr/code-server/issues/2975#issuecomment-809740408

This issue can be reproduced in VS Code: Yes/No

**Comment by jsjoeio** [comment]
@nodesocket do you mind providing the following information so I can update the issue?

```markdown
## OS/Web Information

- Web Browser: Chrome 89.0.4389.90/Firefox 87.0
- Local OS:
- Remote OS:
- Remote Architecture:
- `code-server --version`:

## Steps to Reproduce

1.
2.
3.
```

**Comment by nodesocket** [comment]
## OS/Web Information

- Web Browser: Chrome 89.0.4389.90/Firefox 87.0
- Local OS: MacOS Big Sur
- Remote OS: Docker / Kubernetes using official container 3.9.2.
- Remote Architecture:
- `code-server --version`:

```
coder@code-server-6784d969bd-8w6rb:~$ code-server --version
3.9.2 109d2ce3247869eaeab67aa7e5423503ec9eb859
```

## Steps to Reproduce

1. I upgraded the Kubernetes images from 3.9.1 to 3.9.2.
2. Then logged into my code-server and seeing these `WebSocket close with status code 1006`

**Comment by jsjoeio** [comment]
@bpmct I know you're a Kubernetes guru so if you have any interest in helping with this one, let us know!

**Comment by bpmct** [comment]
@nodesocket are you using the helm chart?

**Comment by nodesocket** [comment]
@bpmct correct, using https://github.com/cdr/code-server/tree/main/ci/helm-chart. I think the problem is not related to Kubernetes, Helm, or Docker, but the `code-server` files itself. See https://github.com/cdr/code-server/issues/2975#issuecomment-813567492

**Comment by nodesocket** [comment]
@bpmct I just started from scratch.

```
helm delete code-server --purge
kubectl delete secret code-server
```

And re-installed _(after confirming it freshly created a new volume)_. To my surprise, I am having the same exact issue. I think something is actually broken in the `3.9.2` source code build.

**Comment by bpmct** [comment]
if you change the image to 3.9.1 in the helm chart do you still run into these issues?

**Comment by Yamilquery** [comment]
I have the same issue with 3.9.2.

I can't downgrade to 3.9.1 because is not available on homebrew.

**Comment by jsjoeio** [comment]
@yamajik you can uninstall with homebrew:
```sh
brew uninstall code-server
```

and reinstall with yarn/npm:
```sh
yarn global add code-server@3.9.1

npm i -g code-server@3.9.1
```

**Comment by nodesocket** [comment]
@bpmct just upgrade my Helmm chart to use `3.9.3` and did a Helm upgrade and things are working now. So, it appears the fix added in `3.9.3` fixed the issue.

**Comment by jsjoeio** [comment]
Awesome! Thanks for checking and letting us know. Going to close this for now. Feel free to open a new issue or post in the v3.9.3 if you see anything funky.
